src/api/client/Network.ts

```diff
@@ -549,4 +549,29 @@ export class Network {
   public async getMiddlewareMetadata(): Promise<MiddlewareMetadata | null> {
     return this.context.getMiddlewareMetadata();
   }
+
+  /**
+   * Check if the middleware is up to date with the chain.
+   *
+   * @note Since the middleware can take some time in processing large blocks,
+   * the function returns true if the number of processed blocks by the middleware doesn't lag the finalized chain blocks by more than 20.
```

Comment:
```suggestion
   * the function returns true if the number of processed blocks by the middleware doesn't lag the finalized chain blocks by more than @link{MAX_MIDDLEWARE_LAG}.
```

I forget the syntax, but the number could become inaccurate easily